src/compiler/transformers/module/module.ts

```diff
@@ -3,16 +3,756 @@
+
+                case SyntaxKind.ImportEqualsDeclaration:
+                    return visitImportEqualsDeclaration(<ImportEqualsDeclaration>node);
+
+                case SyntaxKind.ExportDeclaration:
+                    return visitExportDeclaration(<ExportDeclaration>node);
+
+                case SyntaxKind.ExportAssignment:
+                    return visitExportAssignment(<ExportAssignment>node);
+
+                case SyntaxKind.VariableStatement:
+                    return visitVariableStatement(<VariableStatement>node);
+
+                case SyntaxKind.FunctionDeclaration:
+                    return visitFunctionDeclaration(<FunctionDeclaration>node);
+
+                case SyntaxKind.ClassDeclaration:
+                    return visitClassDeclaration(<ClassDeclaration>node);
+
+                default:
+                    // This visitor does not descend into the tree, as export/import statements
+                    // are only transformed at the top level of a file.
+                    return node;
+            }
+        }
+
+        /**
+         * Visits an ImportDeclaration node.
+         *
+         * @param node The ImportDeclaration node.
+         */
+        function visitImportDeclaration(node: ImportDeclaration): OneOrMany<Statement> {
+            if (contains(externalImports, node)) {
+                const statements: Statement[] = [];
+                const namespaceDeclaration = getNamespaceDeclarationNode(node);
+                if (moduleKind !== ModuleKind.AMD) {
+                    if (!node.importClause) {
+                        // import "mod";
+                        addNode(statements,
+                            createStatement(
+                                createRequireCall(node),
+                                /*location*/ node
+                            )
+                        );
+                    }
+                    else {
+                        const variables: VariableDeclaration[] = [];
+                        if (namespaceDeclaration && !isDefaultImport(node)) {
+                            // import * as n from "mod";
+                            addNode(variables,
+                                createVariableDeclaration(
+                                    getSynthesizedClone(namespaceDeclaration.name),
+                                    createRequireCall(node)
+                                )
+                            );
+                        }
+                        else {
+                            // import d from "mod";
+                            // import { x, y } from "mod";
+                            // import d, { x, y } from "mod";
+                            // import d, * as n from "mod";
+                            addNode(variables,
+                                createVariableDeclaration(
+                                    getGeneratedNameForNode(node),
+                                    createRequireCall(node)

```

Comment:
should either this call to createVariableDeclaration or the next one use `node` as the location?